lib/devise/models/recoverable.rb

```diff
@@ -33,6 +33,8 @@ def self.required_fields(klass)
       # Update password saving the record and clearing token. Returns true if
       # the passwords are valid and the record was saved, false otherwise.
       def reset_password(new_password, new_password_confirmation)
+        self.errors.add(:password, :blank) && return unless new_password
```

Comment:
Please replace this inline check with a full fledged `if`/ `else` around the existing code - there's a lot going on on a single line and it will be more explicit to have the both scenarios handled through the `if`/`else`.